sport/app/football/views/matchList/matchesComponent.scala.html

```diff
@@ -1,25 +1,31 @@
+@import org.joda.time.LocalDate
 @import football.model.MatchesList
+@import model.Competition
+@import pa.FootballMatch
 @import views.support.`package`.Seq2zipWithRowInfo
+@import views.support.RowInfo
 
 @(matchesList: MatchesList, customLink: Option[(String, String)] = None)(implicit request: RequestHeader, context: model.ApplicationContext)
 
-<div data-component="football-matches-embed" class="c-football-matches football-embed">
+@template1(date: LocalDate, competitionMatches: List[(Competition, List[FootballMatch])], dateRow: RowInfo) = {
+    <div class="football-matches__day">
+    @{competitionMatches.zipWithRowInfo.map { case ((competition, matches), matchRow) =>
+        football.views.html.matchList.matchesList(
+            matches,
+            competition,
+            date,
+            matchType = matchesList.pageType,
+            heading = if(dateRow.isFirst) Some((competition.fullName, Option(competition.url))) else None,
+            link = customLink orElse (if(dateRow.isLast && matchRow.isLast)
+                Some(("View all "+matchesList.pageType, "/football/"+matchesList.pageType)) else None)
+        )
+    }}
+    </div>
+}
 
+<div data-component="football-matches-embed" class="c-football-matches football-embed">
     @football.views.html.fragments.componentStyles()
-
-    @matchesList.matchesGroupedByDateAndCompetition.zipWithRowInfo.map { case ((date, competitionMatches), dateRow) =>
-        <div class="football-matches__day">
-            @competitionMatches.zipWithRowInfo.map { case ((competition, matches), matchRow) =>
-                @football.views.html.matchList.matchesList(
-                    matches,
-                    competition,
-                    date,
-                    matchType = matchesList.pageType,
-                    heading = if(dateRow.isFirst) Some((competition.fullName, Option(competition.url))) else None,
-                    link = customLink orElse (if(dateRow.isLast && matchRow.isLast)
-                            Some(("View all "+matchesList.pageType, "/football/"+matchesList.pageType)) else None)
-                )
-            }
-        </div>
+    @matchesList.matchesGroupedByDateAndCompetition.zipWithRowInfo.map { data =>
+        @template1(data._1._1, data._1._2, data._2)
```

Comment:
Nested tuple 😢 

If not too hard we could add a type for `matchesGroupedByDateAndCompetition` where it's defined elsewhere which names `date` and `competitionMatches`, then it's just the one tuple we're dealing with. Otherwise, we could use a `@defining` block here to name things?

Reply:
So yeah, I was focused on fixing the variable scope problem and told myself not to do anything not needed beyond just that, but I was very tempted to make a case class to simplifying things. 

I tried `@defining` blocks, they actually work very well at resolving the variable scope problem, but they made the code look even more like a PHP-kind mix between Scala code and HTML. So I decided that, all things considered, extracting to a separate template (adding a new function) was the lesser of two evils.